### ISSUE TITLE
Orderedset can't be installed without C++?

### ISSUE THREAD

**Issue by DenisCarriere** [DenisCarriere/geocoder#270]
@ebreton Strange, can't install `geocoder` anymore on Windows.

The root of the problem seems to be coming from `orderedset`.

## Solution

Isn't a native Python way to do this or can we simply include the Python code directly instead of relying on an external unstable dependency.

Bundle this internally?
https://github.com/LuminosoInsight/ordered-set/blob/master/ordered_set.py

## Logs

```
(env) C:\Users\Owner\Documents\GitHub>pip install geocoder
Collecting geocoder
  Using cached geocoder-1.25.0-py2.py3-none-any.whl
Collecting six (from geocoder)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting click (from geocoder)
  Using cached click-6.7-py2.py3-none-any.whl
Collecting orderedset (from geocoder)
  Using cached orderedset-2.0.tar.gz
Collecting requests (from geocoder)
  Downloading requests-2.18.3-py2.py3-none-any.whl (88kB)
    100% |################################| 92kB 692kB/s
Collecting ratelim (from geocoder)
  Using cached ratelim-0.1.6-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests->geocoder)
  Downloading chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |################################| 143kB 939kB/s
Collecting certifi>=2017.4.17 (from requests->geocoder)
  Downloading certifi-2017.7.27.1-py2.py3-none-any.whl (349kB)
    100% |################################| 358kB 2.3MB/s
Collecting idna<2.6,>=2.5 (from requests->geocoder)
  Downloading idna-2.5-py2.py3-none-any.whl (55kB)
    100% |################################| 61kB 5.1MB/s
Collecting urllib3<1.23,>=1.21.1 (from requests->geocoder)
  Downloading urllib3-1.22-py2.py3-none-any.whl (132kB)
    100% |################################| 133kB 3.8MB/s
Collecting decorator (from ratelim->geocoder)
  Using cached decorator-4.1.2-py2.py3-none-any.whl
Building wheels for collected packages: orderedset
  Running setup.py bdist_wheel for orderedset ... error
  Complete output from command c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-4zwkwo\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d c:\users\owner\appdata\local\temp\tmpgzknlgpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win32-2.7
  creating build\lib.win32-2.7\orderedset
  copying lib\orderedset\__init__.py -> build\lib.win32-2.7\orderedset
  running build_ext
  building 'orderedset._orderedset' extension
  error: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27

  ----------------------------------------
  Failed building wheel for orderedset
  Running setup.py clean for orderedset
Failed to build orderedset
Installing collected packages: six, click, orderedset, chardet, certifi, idna, urllib3, requests, decorator, ratelim, geocoder
  Running setup.py install for orderedset ... error
    Complete output from command c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-4zwkwo\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\owner\appdata\local\temp\pip-qcrcss-record\install-record.txt --single-version-externally-managed --compile --install-headers c:\users\owner\documents\github\env\include\site\python2.7\orderedset:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win32-2.7
    creating build\lib.win32-2.7\orderedset
    copying lib\orderedset\__init__.py -> build\lib.win32-2.7\orderedset
    running build_ext
    building 'orderedset._orderedset' extension
    error: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27

    ----------------------------------------
Command "c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-4zwkwo\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\owner\appdata\local\temp\pip-qcrcss-record\install-record.txt --single-version-externally-managed --compile --install-headers c:\users\owner\documents\github\env\include\site\python2.7\orderedset" failed with error code 1 in c:\users\owner\appdata\local\temp\pip-build-4zwkwo\orderedset\

(env) C:\Users\Owner\Documents\GitHub>
(env) C:\Users\Owner\Documents\GitHub>pip install orderedset
Collecting orderedset
  Using cached orderedset-2.0.tar.gz
Building wheels for collected packages: orderedset
  Running setup.py bdist_wheel for orderedset ... error
  Complete output from command c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-ghoklv\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d c:\users\owner\appdata\local\temp\tmppkbv7apip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win32-2.7
  creating build\lib.win32-2.7\orderedset
  copying lib\orderedset\__init__.py -> build\lib.win32-2.7\orderedset
  running build_ext
  building 'orderedset._orderedset' extension
  error: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27

  ----------------------------------------
  Failed building wheel for orderedset
  Running setup.py clean for orderedset
Failed to build orderedset
Installing collected packages: orderedset
  Running setup.py install for orderedset ... error
    Complete output from command c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-ghoklv\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\owner\appdata\local\temp\pip-mxgmdh-record\install-record.txt --single-version-externally-managed --compile --install-headers c:\users\owner\documents\github\env\include\site\python2.7\orderedset:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win32-2.7
    creating build\lib.win32-2.7\orderedset
    copying lib\orderedset\__init__.py -> build\lib.win32-2.7\orderedset
    running build_ext
    building 'orderedset._orderedset' extension
    error: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27

    ----------------------------------------
Command "c:\users\owner\documents\github\env\scripts\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\owner\\appdata\\local\\temp\\pip-build-ghoklv\\orderedset\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\owner\appdata\local\temp\pip-mxgmdh-record\install-record.txt --single-version-externally-managed --compile --install-headers c:\users\owner\documents\github\env\include\site\python2.7\orderedset" failed with error code 1 in c:\users\owner\appdata\local\temp\pip-build-ghoklv\orderedset\
```

**Comment by ebreton** [comment]
aie. yes the orderedset is indeed messing around.

Well, maybe the easiest solution would be to replace them with basic arrays.

My idea behind orderedset was to take advantage of the functionnalities of a 'set', i.e union, intersection, and so on... but we are not using any of this right now. So let's go for the simplest way of doing things.

:)

I will push a PR in the next days if not by tomorrow

**Comment by DenisCarriere** [comment]
👍 Awesome, thanks! Definitely need to avoid C++ / compiled dependencies. Python has `OrderedDict`, strange it doesn't have the same for sets... ah well.. maybe a flat Array would be simplest.

Thanks for looking into this.

**Comment by ebreton** [comment]
yep, I am using OrderedDict quite often, but the set would have been better to manipulate the results...

anyway, let's KIS ^^

**Comment by ebreton** [comment]
Fixed in PR #271 

If that's good for you, please to the pypi release (haven't created my account yet)

**Comment by DenisCarriere** [comment]
👍 Sweet it works now! Bumped the release to `1.26.0`
